flake.nix

```diff
@@ -0,0 +1,63 @@
+{
+  description = "A Nix-flake-based development environment for Cursorless";
+
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+
+  outputs =
+    { self, nixpkgs }:
+    let
+      supportedSystems = [
+        "x86_64-linux"
+        "aarch64-linux"
+        "x86_64-darwin"
+        "aarch64-darwin"
+      ];
+      forEachSupportedSystem =
+        f: nixpkgs.lib.genAttrs supportedSystems (system: f { pkgs = import nixpkgs { inherit system; }; });
+      pythonVersion = builtins.replaceStrings [ "py" ] [
+        "python"
+      ] (nixpkgs.lib.importTOML ./pyproject.toml).tool.ruff.target-version;
+    in
+    {
+      devShells = forEachSupportedSystem (
+        { pkgs }:
+        {
+          default = pkgs.mkShell {
+            packages =
+              let
+                python = pkgs.${pythonVersion};
+                pythonPackages = pkgs."${pythonVersion}Packages";
+              in
+              [
+                pkgs.corepack
+                pkgs.vsce
+                pkgs.pre-commit
+
+                python
+                # Dependencies from pre-commit python scripts
+                # run `pre-commit run --all-files` to find out what you need.
+                pythonPackages.identify
+                pythonPackages.cfgv
+                pythonPackages.yamllint
+              ];
+            # To prevent weird broken non-interactive bash terminal
+            buildInputs = [ pkgs.bashInteractive ];
+            shellHook = ''
+              if [ ! -f .git/hooks/pre-commit ]; then
+                pre-commit install
+              fi
+
+              # Cursorless .prettierrc depends on prettier-plugin-tailwindcss, which
+              # pre-commit won't auto-install. So we have to manually install it into the
+              # workspace until there is a nix package.
+              if ! pnpm ls prettier-plugin-tailwindcss | grep prettier >/dev/null; then
+                pnpm -w -D install prettier-plugin-tailwindcss
+                # This updates the pnpm-lock.yaml, which we don't want to keep
+                git restore pnpm-lock.yaml
```

Comment:
If the developer had modified the pnpm lock before this happens, does it get clobbered? That doesn't seem great if so

Reply:
Yeah this doesn't seem like a great idea. If we really want to install pre-commit each time a `nix shell` is started then it'd make more sense to simply do a `pnpm install`

Reply:
Ya makes sense. I updated it to `pnpm install` now. @auscompgeek I assume by 'install pre-commit' you mean `prettier-plugin-tailwindcss?` Running `pnpm install` doesn't run `pre-commit install` afaik, so that part needs to stay the same.